Image content grid block

nimithshetty17 opened the pull request:
## Issue

Fixes #[MERATIVE-910](https://merative.atlassian.net/browse/MERATIVE-910)

## Description

Creation of the image content grid block.

- Figma Link - https://www.figma.com/file/vw24IPXXt4vCh9BNumka9A/Web--Merative-Digital-Design-System-2.0?type=design&node-id=21610-14427&mode=design&t=uDzjrfRhEKyqDCEm-0

## Test URLs
  
- Before (Changes from `main`): https://main--merative2--nimithshetty17.hlx.page/block-library/blocks/image-content-grid
- After (Changes from this PR): https://image-content-grid-block--merative2--nimithshetty17.hlx.page/block-library/blocks/image-content-grid

sachinmesh reviewed (changes requested):
1. Spacing between the list heading and paragrpagh below it should be 16px and not 24px. 
<img width="275" alt="image" src="https://github.com/hlxsites/merative2/assets/124026684/eb5db86f-993e-4b0b-8919-07dc8a3757a7">

2. The list container size is wrong 
<img width="381" alt="image" src="https://github.com/hlxsites/merative2/assets/124026684/d785c38d-d67d-49fe-b2c7-0b435c0980c2">


3. The image container size is wrong. The image and the text on the right should have 120px spacing 
<img width="397" alt="image" src="https://github.com/hlxsites/merative2/assets/124026684/6933520c-67f9-4784-be0b-a777e9a8c422">

4. On mobile, the spacing between the list heading and paragraph below should be 16px and not 24px. 
<img width="354" alt="image" src="https://github.com/hlxsites/merative2/assets/124026684/532a9b4c-1c2c-4950-aab8-be2048e6967b">


5. waiting for the bone background version also.

nimithshetty17 commented:
@sachinmesh please find the fixes below:-

1. Spacing between the list heading and paragraph below it should be 16px and not 24px. - Fixed
2. The list container size is wrong - Fixed
3. The image container size is wrong. The image and the text on the right should have 120px spacing. - Fixed
4. On mobile, the spacing between the list heading and paragraph below should be 16px and not 24px. -Fixed
5. Waiting for the bone background version also. - Tuned image to look half circle even in the white background.

sachinmesh commented:
1. These list containers should be of 305 width 
<img width="444" alt="image" src="https://github.com/hlxsites/merative2/assets/124026684/c4f86e3a-ee16-4662-845f-8faef32c392b">

2. The semicircle image dimensions are wrong.. Check the correct one
<img width="599" alt="image" src="https://github.com/hlxsites/merative2/assets/124026684/111741e2-dd54-4e3b-a637-b4684a381a32">

Sharing the figma file here.  https://www.figma.com/file/vw24IPXXt4vCh9BNumka9A/Web--Merative-Digital-Design-System-2.0?type=design&node-id=21610-14449&mode=design&t=D6dm8a9w1cfpxcQ3-4

@nimithshetty17 @anabarcelona @keith-kaplan

sachinmesh commented:
Good to go @nimithshetty17 @keith-kaplan @anabarcelona

keith-kaplan reviewed (approved):
Looks great.
